### PR TITLE
Adds checklists for doing initial PR review.

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

File: docs/source/developer/contributing.rst [hunk 1]
```diff
@@ -422,6 +439,145 @@ as to free up core developer time. Examples of ways to help:
...
+        * [ ] The implementation has references to the region of the NumPy code
+              base that provides the same functionality. The references are URLs
+              of the form:
+              `https://github.com/numpy/numpy/tree/<SHA>/path/to/file.py:#<lines>`
+        * [ ] The code "looks" reasonably similar to the NumPy implementation.
+        * [ ] The code uses the same algorithm as the NumPy implementation, if
+              it does not it explains why and this justification is reasonable.
+        * [ ] The type inference part of the Numba implementation restricts the
+              types to those declared in the NumPy docs (or if not present, what
+              is accepted in the NumPy implementation itself).
```
Comment (esc) [inline docs/source/developer/contributing.rst:485]: Probably worth adding that those restrictions are tested by supplying bogus types and checking that the error message is indeed the expected one.

File: docs/source/developer/contributing.rst [hunk 2]
```diff
@@ -422,6 +439,145 @@ as to free up core developer time. Examples of ways to help:
...
+          please highlight this and one of the core developers will pick it up.)
+    * [ ] The proposed additional NumPy functionality is not deprecated by
+          NumPy.
+    * [ ] Public CI is passing.
+    * [ ] The patch does not touch files in `numba.core`.
+    * [ ] The patch is implemented using the high-level extension API only
+          (https://numba.readthedocs.io/en/stable/extending/high-level.html).
+    * [ ] The patch updates the documentation to reflect the new functionality.
+          (https://github.com/numba/numba/blob/main/docs/source/reference/numpysupported.rst)
+    * [ ] The patch is line wrapped to 80 characters.
```
Comment (stuartarchibald) [inline docs/source/developer/contributing.rst:471]: Need to add a note about running a "spell checker", and making sure that NumPy/SciPy/Python are correctly capitalised.

Comment (sklam) [reply docs/source/developer/contributing.rst:471]: hm... can we automate these type of checks as part of CI?

File: docs/source/developer/contributing.rst [hunk 3]
```diff
@@ -422,6 +439,145 @@ as to free up core developer time. Examples of ways to help:
...
+              them to track their origin. The references are URLs of the form:
+              `https://github.com/numpy/numpy/tree/<SHA>/path/to/file.py:#<lines>`
+        * [ ] Use `unittest` style assertions (not pytest style!).
+        * [ ] Check that unsupported/invalid input types are handled as expected
+              using the `with self.assertRaises()` pattern.
+        * [ ] Explicitly check custom error messages are raised appropriately.
+        * [ ] Hit all the branches of the proprosed change (both at type
+              inference and run time).
+        * [ ] Check that numerically awkward values work as expected e.g.
+              `NaN`, `+/-Inf`, `0`, `INT_MAX`, `INT_MIN` etc.
```
Comment (sklam) [inline docs/source/developer/contributing.rst:508]: I wonder if we should reference Boundary Value Testing in addition to this as this is a subset of BVT. BVT will also include empty containers.

**Review by sklam — COMMENTED** [review]
This is very thorough, but i worry that it is so long it can be overwhelming. I don't think we need to mention things that are already automated. As for testing, would it be better if there's a link/page to explain our testing standard?

File: docs/source/developer/contributing.rst [hunk 4]
```diff
@@ -422,6 +439,145 @@ as to free up core developer time. Examples of ways to help:
 
 The core developers thank everyone for their understanding with the above!
 
+.. _reviewing_pull_requests:
+
+Reviewing Pull Requests
+-----------------------
+
+Help with reviewing pull requests is welcomed. There are time-effort categorised
+review tasks available on this project board: **TODO: ADD LINK**.
```
Comment (esc) [inline docs/source/developer/contributing.rst:448]:
```suggestion
review tasks available on the `Contributor self-service board <https://github.com/orgs/numba/projects/7>`_.
```

**Review by esc — CHANGES_REQUESTED** [review]
Adding the link to the contributor self-service board.

**Pull request closed without merging**